warehouse/query-core/src/test/java/datawave/query/tables/RangeStreamScannerTest.java

```diff
@@ -0,0 +1,374 @@
+     *
+     * This version creates a simple iterator that will not alter the index records coming back in any way.
+     *
+     * @param fieldName
+     *            - field name, like "FOO" in "FOO == 'bar'"
+     * @param fieldValue
+     *            - field value, like "bar" in "FOO == 'bar'"
+     * @return a configured RangeStreamScanner
+     */
+    private RangeStreamScanner buildRangeStreamScanner(String fieldName, String fieldValue) throws Exception {
+        
+        String queryString = fieldName + "=='" + fieldValue + "'";
+        
+        // Build the executors
+        int maxLookup = (int) Math.max(Math.ceil(config.getNumIndexLookupThreads()), 1);
+        BlockingQueue<Runnable> runnables = new LinkedBlockingDeque<>();
+        int executeLookupMin = Math.max(maxLookup / 2, 1);
+        ExecutorService streamExecutor = new ThreadPoolExecutor(executeLookupMin, maxLookup, 100, TimeUnit.MILLISECONDS, runnables);
+        
+        int priority = 50; // Iterator priority
+        
+        // Build options for RangeStreamScanner
+        SessionOptions options = new SessionOptions();
+        options.fetchColumnFamily(new Text(fieldName));
+        
+        // Set iterator datatype filter
+        IteratorSetting is = new IteratorSetting(priority++, DataTypeFilter.class);
+        is.addOption(DataTypeFilter.TYPES, config.getDatatypeFilterAsString());
+        options.addScanIterator(is);
+        
+        // Set iterator option. Do not collapse uids into shard ranges, just pass results back.
+        final IteratorSetting uidSetting = new IteratorSetting(priority++, CreateUidsIterator.class);
+        uidSetting.addOption(CreateUidsIterator.COLLAPSE_UIDS, Boolean.valueOf(false).toString());
+        options.addScanIterator(uidSetting);
+        
+        // Set query option
+        options.addScanIterator(QueryScannerHelper.getQueryInfoIterator(config.getQuery(), false, queryString));
+        
+        // Build RangeStreamScanner
+        RangeStreamScanner rangeStreamScanner = scannerFactory.newRangeScanner(config.getIndexTableName(), config.getAuthorizations(), config.getQuery(),
+                        config.getShardsPerDayThreshold());
+        rangeStreamScanner.setMaxResults(config.getMaxIndexBatchSize());
+        rangeStreamScanner.setExecutor(streamExecutor);
+        rangeStreamScanner.setRanges(Collections.singleton(rangeForTerm(fieldValue, fieldName, config))).setOptions(options);
+        
+        return rangeStreamScanner;
+    }
+    
+    /**
+     * Make sure a simple scan returns correctly. FOO == 'bar' hits day 20190314 with 1 shard, each shard has 2 document ids.
+     */
+    @Test
+    public void testTheSimplestOfScans() throws Exception {
+        
+        // Components that define the query: "FOO == 'bar'"
+        String fieldName = "FOO";
+        String fieldValue = "bar";
+        ASTEQNode eqNode = (ASTEQNode) JexlNodeFactory.buildEQNode(fieldName, fieldValue);
```

Comment:
You removed a similar cast in EntryParser.  Where do you really stand on unnecessary casts?  :)

Reply:
You'll never know!

Reply:
Ah, I missed that!  OK, well everything looks good to me now.